### PR TITLE
Migrate from CRA to Vite

### PR DESCRIPTION
**Pull request by garg3133**
Since CRA is no longer maintained, this PR aims to migrate this project from using `react-scripts` to Vite. Also, migrates the project to ESM (`type=module`).

Doing so also fixes almost all the issues reported in #69 with an exception of the following deprecation warnings:
```sh
> npm i
             
npm WARN deprecated @material-ui/styles@4.11.5: Material UI v4 doesn't receive active development since September 2021. See the guide https://mui.com/material-ui/migration/migration-v4/ to upgrade to v5.
npm WARN deprecated @material-ui/pickers@3.3.11: This package no longer supported. It has been relaced by @mui/x-date-pickers
npm WARN deprecated @material-ui/lab@4.0.0-alpha.61: Material UI v4 doesn't receive active development since September 2021. See the guide https://mui.com/material-ui/migration/migration-v4/ to upgrade to v5.
npm WARN deprecated @material-ui/core@4.12.4: Material UI v4 doesn't receive active development since September 2021. See the guide https://mui.com/material-ui/migration/migration-v4/ to upgrade to v5.
```

After this change, the website continues to work as before, but is yet to be tested on production (though running `npm run serve` after building the project works fine).

Guide followed during this migration: https://www.freecodecamp.org/news/how-to-migrate-from-create-react-app-to-vite/

### Next Steps:
* Upgrade the packages that have backward compatibility issues (like `react-router-dom` and `@material-ui/core`) before upgrading React to `v18` since the latest versions of these packages support React v17 as well.
* Upgrade React to `v18`.

### CODE REVIEW

File: frontend/index.html [hunk 1]
```diff
@@ -2,7 +2,6 @@
 <html lang="en">
   <head>
     <meta charset="utf-8" />
-    <link rel="icon" href="./bitbyte.png" />
```
Comment (Aksh-Bansal-dev) [inline frontend/index.html:5]: I think you removed the icon by mistake.

**Review by Aksh-Bansal-dev — COMMENTED** [review]
Everything else seems to be working fine.

File: frontend/index.html (hunk 1 above)
Comment (garg3133) [reply frontend/index.html:5]: This was a duplicate line, the icon is still present [here](https://github.com/BitByte-TPC/TPC-website/blob/376dab9696795d5f993e5882526b771384b5c760/frontend/index.html#L22-L26). And moreover, there is no file named `bitbyte.png`.

Comment (Aksh-Bansal-dev) [reply frontend/index.html:5]: The icon was not showing when I ran it in local. Maybe we have to change the path of the icon [here](https://github.com/BitByte-TPC/TPC-website/blob/376dab9696795d5f993e5882526b771384b5c760/frontend/index.html#L22-L26) to `/public/favicon.ico`

Comment (garg3133) [reply frontend/index.html:5]: It was working fine for me, but I've still changed the path to `/favicon.ico`. We don't need to prepend `public` to the path as Vite by itself looks for static assets in the `public` directory by default, which is configurable using the `publicDir` vite config: https://vitejs.dev/config/shared-options.html#publicdir

**Review by Aksh-Bansal-dev — APPROVED** [review]
Looks good.

**Pull request merged**